### PR TITLE
Handle unicode in exception messages

### PR DESCRIPTION
**Pull request by bwbeach**
Fix for resurrected #115

### CODE REVIEW

File: b2/exception.py [hunk 1]
```diff
@@ -10,13 +10,28 @@
...
+        """
+        Python 2 does not like it when you pass unicode as the message
+        in an exception.  We like to use file names in exception messages.
+        To avoid problems, if the message has any non-ascii characters in
+        it, they are replaced with backslash-uNNNN
+
+        https://pythonhosted.org/kitchen/unicode-frustrations.html#frustration-5-exceptions
+        """
+        if six.PY2:
+            if 0 < len(args) and isinstance(args[0], six.text_type):
```
Comment (ppolewicz) [inline b2/exception.py]: we usually avoid `<`. How about this:
```python
if args and isinstance(args[0], six.text_type):
```

File: b2/exception.py [hunk 2]
```diff
@@ -10,13 +10,28 @@
...
+                args = tuple([json.dumps(args[0])[1:-1]] + list(args[1:]))
```
Comment (ppolewicz) [inline b2/exception.py]: there has to be a better way to encode it than using json

File: b2/exception.py (hunk 1 above)
Comment (bwbeach) [inline b2/exception.py]: works for me.

File: b2/exception.py (hunk 2 above)
Comment (bwbeach) [inline b2/exception.py]: This was quick and easy.  

I could simple write the code to replace chars above 127 with a `\uNNNN` code.  Would you like that better?

Comment (ppolewicz) [inline b2/exception.py]: I found out how the [json implementation](https://github.com/python/cpython/blob/master/Lib/json/encoder.py) looks like. They even have it optimized in c... So lets keep your current implementation.

**Pull request merged**